### PR TITLE
Add logging for instrumentation lifecycle methods

### PR DESCRIPTION
**Pull request by marcelogomez**
This PR does some refactoring of `DefaultServerInstrumentation` and `JUnitRunnerInstrumentation` to reduce redundancies and adds some log statements for the instrumentation's lifecycle

### CODE REVIEW

File: selendroid-server/src/main/java/io/selendroid/server/DefaultServerInstrumentation.java [hunk 1]
```diff
@@ -60,6 +60,13 @@ public DefaultServerInstrumentation(Instrumentation instrumentation,
         this.instrumentation = instrumentation;
         this.args = args;
 
+        // Get a wake lock to stop the cpu going to sleep
+        PowerManager pm =(PowerManager) instrumentation
```
Comment (asm89) [inline selendroid-server/src/main/java/io/selendroid/server/DefaultServerInstrumentation.java]: nit: space

Comment (asm89) [inline selendroid-server/src/main/java/io/selendroid/server/DefaultServerInstrumentation.java]: Should we be cautious here?

File: selendroid-server/src/main/java/io/selendroid/server/DefaultServerInstrumentation.java [hunk 2]
```diff
@@ -146,12 +148,36 @@ public void startActivity(String activityClassName) {
 
     @Override
     public void startServer() {
+        SelendroidLogger.info("*** ServerInstrumentation#startSever() ***");
```
Comment (asm89) [inline selendroid-server/src/main/java/io/selendroid/server/DefaultServerInstrumentation.java]: nit: `startServer`

File: selendroid-server/src/main/java/io/selendroid/server/DefaultServerInstrumentation.java [hunk 3]
```diff
@@ -146,12 +148,36 @@ public void startActivity(String activityClassName) {
...
+                  SelendroidLogger.error("Failed to start Selendroid server", e);
+              }
+          }
+        });
+    }
+
+    // TODO: Make this class a base class and extract all the serverthread
+    // specific logic to a subclass
+    protected void startServerImpl() {
+        SelendroidLogger.info("*** ServerInstrumentation#startSeverImpl() ***");
```
Comment (asm89) [inline selendroid-server/src/main/java/io/selendroid/server/DefaultServerInstrumentation.java]: s/Sever/Server

File: selendroid-server/src/main/java/io/selendroid/server/DefaultServerInstrumentation.java (hunk 1 above)
Comment (marcelogomez) [inline selendroid-server/src/main/java/io/selendroid/server/DefaultServerInstrumentation.java]: What do you mean? `null` check? I just moved this from somewhere else

Comment (asm89) [inline selendroid-server/src/main/java/io/selendroid/server/DefaultServerInstrumentation.java]: Let's keep it like this for now then. Just caught my eye.

**Pull request closed without merging**